### ISSUE TITLE
Disk Utilization related to suddent NaN loss after a few epochs?

### ISSUE THREAD

**Issue by mikedataCrunch** [ultralytics/ultralytics#19944]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi, beginner here. I'm training models in a loop wherein each loop is a new data.yaml and a new init of model = YOLO(...). I noticed that there is patterin of sudden drops then linear increase in my Disk Utilization. Wondering if this is normal or is some unpreferred set behavior that is linked to my loss dropping to NaNs after a few epochs for most of the runs in the loop. 

![Image](https://github.com/user-attachments/assets/b15eec6f-ab22-4ad7-9bb5-001bb8122c00)
![Image](https://github.com/user-attachments/assets/a4dc2326-b797-46e5-ae94-76a20a3dfb30)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @mikedataCrunch, thank you for reaching out and for your interest in Ultralytics 🚀! 

We recommend checking out the [Docs](https://docs.ultralytics.com/) for detailed guidance. You can find many examples for [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage, which might help clarify or resolve your question.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate the issue further. This will help us understand the behavior you're observing with disk utilization and NaN loss.

If this is a ❓ Question related to custom training, please share more details such as:
- Your training setup (hardware, batch size, etc.)
- A snippet of your dataset's `data.yaml`
- Any training logs or additional context
- Example images that might illustrate the issue

In the meantime, ensure you're using the latest version of the `ultralytics` package and its dependencies by upgrading in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

### Environments
For your convenience, YOLO is verified to run in the following environments (all dependencies pre-installed):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

You can also connect with the Ultralytics community for additional support:
- Join our [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time conversations
- Visit our [Discourse](https://community.ultralytics.com/) for in-depth discussions
- Check out our [Subreddit](https://reddit.com/r/Ultralytics) for community-driven insights

### Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing, confirming the integrity of YOLO across all supported platforms.

This is an automated response to guide you while an Ultralytics engineer reviews your issue shortly. Thank you for your patience and for helping us improve YOLO 😊!

**Comment by glenn-jocher** [comment]
Hi! Sudden NaN loss can be linked to disk caching behavior during training. The Ultralytics YOLO data loader attempts to cache images to disk as *.npy files for faster loading, which might explain the disk utilization spikes. If disk space runs low during this process, it could lead to incomplete caching or data corruption, potentially contributing to NaN losses. 

We recommend:
1. Checking free disk space with our [check_cache_disk() function](https://docs.ultralytics.com/reference/data/base/#ultralytics.data.base.BaseDataset.check_cache_disk)
2. Monitoring your dataset integrity (corrupted images/labels often cause NaN loss)
3. Temporarily disabling disk caching with `cache=False` in your train arguments

For more details, see our [Troubleshooting Common YOLO Issues guide](https://docs.ultralytics.com/guides/yolo-common-issues/). Let us know if this helps! 🚀
